modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptReduxQueryClientCodegen.java

```diff
@@ -0,0 +1,292 @@
+        this.modelPackage = "src" + File.separator + "models";
+        this.apiTemplateFiles.put("apis.mustache", ".ts");
+        this.modelTemplateFiles.put("models.mustache", ".ts");
+        this.addExtraReservedWords();
+
+        this.cliOptions.add(new CliOption(NPM_REPOSITORY, "Use this property to set an url your private npmRepo in the package.json"));
+        this.cliOptions.add(new CliOption(WITH_INTERFACES, "Setting this property to true will generate interfaces next to the default class implementations.", SchemaTypeUtil.BOOLEAN_TYPE).defaultValue(Boolean.FALSE.toString()));
+        this.cliOptions.add(new CliOption(USE_SINGLE_REQUEST_PARAMETER, "Setting this property to true will generate functions with a single argument containing all API endpoint parameters instead of one argument per parameter.", SchemaTypeUtil.BOOLEAN_TYPE).defaultValue(Boolean.TRUE.toString()));
+    }
+
+    @Override
+    public String getName() {
+        return "typescript-redux-query";
+    }
+
+    @Override
+    public String getHelp() {
+        return "Generates a TypeScript client library using redux-query API (beta).";
+    }
+
+    public String getNpmRepository() {
+        return npmRepository;
+    }
+
+    public void setNpmRepository(String npmRepository) {
+        this.npmRepository = npmRepository;
+    }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+        additionalProperties.put("isOriginalModelPropertyNaming", getModelPropertyNaming().equals("original"));
+        additionalProperties.put("modelPropertyNaming", getModelPropertyNaming());
+        supportingFiles.add(new SupportingFile("index.mustache", "src", "index.ts"));
+        supportingFiles.add(new SupportingFile("runtime.mustache", "src", "runtime.ts"));
+        supportingFiles.add(new SupportingFile("tsconfig.mustache", "", "tsconfig.json"));
+        supportingFiles.add(new SupportingFile("gitignore", "", ".gitignore"));
+
+        if (additionalProperties.containsKey(USE_SINGLE_REQUEST_PARAMETER)) {
+            this.setUseSingleRequestParameter(convertPropertyToBoolean(USE_SINGLE_REQUEST_PARAMETER));
+        }
+        writePropertyBack(USE_SINGLE_REQUEST_PARAMETER, getUseSingleRequestParameter());
+
+        if (additionalProperties.containsKey(NPM_NAME)) {
+            addNpmPackageGeneration();
+        }
+    }
+
+    @Override
+    public String getTypeDeclaration(Schema p) {
+        Schema inner;
+        if (ModelUtils.isArraySchema(p)) {
+            inner = ((ArraySchema) p).getItems();
+            return this.getSchemaType(p) + "<" + this.getTypeDeclaration(inner) + ">";
+        } else if (ModelUtils.isMapSchema(p)) {
+            inner = ModelUtils.getAdditionalProperties(p);
+            return "{ [key: string]: " + this.getTypeDeclaration(inner) + "; }";
+        } else if (ModelUtils.isFileSchema(p)) {
+            return "Blob";
+        } else if (ModelUtils.isBinarySchema(p)) {
```

Comment:
overriding with a custom `--type-mappings date=string,date-time=string` would not work here I guess, see also https://github.com/OpenAPITools/openapi-generator/issues/3842

Reply:
Exploring this. I can definitely see the allure of avoiding the JS `Date` and leveraging something like `date-fns` or `moment.js` instead.

Adding that parameter definitely doesn't work, and still produce `Date` typed properties on the various models.

I would ask if this is in scope of this initial PR, or if you'd be ok w/ merging without this pieces working, and then a follow up PR to add this ability.

Reply:
Its ok, I just wanted to point out a potential problem.

Reply:
Yeah, it definitely is, and one I would like to see addressed in the generator at some point.